azurelinuxagent/ga/exthandlers.py

```diff
@@ -484,16 +485,17 @@ def handle_enable(self, ext_handler_i):
             if old_ext_handler_i is None:
                 ext_handler_i.install()
             elif ext_handler_i.version_ne(old_ext_handler_i):
-                ExtHandlersHandler._update_extension_handler(old_ext_handler_i, ext_handler_i)
+                uninstall_failed = ExtHandlersHandler._update_extension_handler_and_return_if_failed(
+                    old_ext_handler_i, ext_handler_i)
         else:
             ext_handler_i.update_settings()
 
-        ext_handler_i.enable()
+        ext_handler_i.enable(uninstall_failed=uninstall_failed)
 
     @staticmethod
-    def _update_extension_handler(old_ext_handler_i, ext_handler_i):
+    def _update_extension_handler_and_return_if_failed(old_ext_handler_i, ext_handler_i):
 
-        def execute_old_handler_command(func):
+        def execute_old_handler_command_and_return_if_failed(func):
```

Comment:
Can you add a docstring for the return value and explain that `True` means it failed? Just by looking at it, I initially thought `True` indicated success as it usually does.

Reply:
Makes sense, I'll modify the doc

Reply:
Changed the function definition to return True on success